lib/email-addresses.js

```diff
@@ -782,19 +796,35 @@ function parse5322(opts) {
         return result;
     }
 
-    function giveResult(ast) {
+    function _giveResult (ast, types) {
         function grabSemantic(n) {
             return n !== null ? n.semantic : null;
-        }
-        var i, ret, addresses, addr, name, aspec, local, domain, comments;
+        }        
+        var i, ret, addresses, addr, name, aspec, local, domain, comments, group_addresses;
         if (ast === null) {
             return null;
         }
         ret = { ast: ast };
-        addresses = findAllNodes('address', ast);
+        addresses = [].concat.apply([], types.map(function (type) { return findAllNodes(type, ast) }));
         ret.addresses = [];
         for (i = 0; i < addresses.length; i += 1) {
             addr = addresses[i];
+            if (addr.seen) continue;
+            addr.seen = true;
+            if (addr.name === 'group') {
```

Comment:
Both achieve things the same way, the "correct" way would be to know just how deep it's OK to go in the AST, IMHO. That way you could guarantee that looking for mailboxes under a group wouldn't happen.